### DISCUSSION TITLE
Next.js Scripts Error: Cannot find module '../../webpack-runtime.js'

### DISCUSSION THREAD

**Discussion by deadcoder0904** [vercel/next.js#27942] [Help]
I want to create RSS script using Next.js.

So I put up a script in a subfolder inside the root folder `scripts/` & named it `build-rss.js`

# next.config.js

```js
module.exports = {
	webpack: (config, options) => {
		config.module.rules.push({
			test: /\.svg$/,
			issuer: { and: [/\.(js|ts|md)x?$/] },
			use: [
				{
					loader: '@svgr/webpack',
					options: {
						prettier: false,
						svgo: true,
						svgoConfig: { plugins: [{ removeViewBox: false }] },
						titleProp: true,
					},
				},
			],
		})

		if (!options.dev && options.isServer) {
			const originalEntry = config.entry

			config.entry = async () => {
				const entries = { ...(await originalEntry()) }
				entries['./scripts/build-rss'] = './scripts/build-rss.js'
				return entries
			}
		}

		if (!options.isServer) {
			config.resolve.fallback.fs = false
		}

		return config
	},
}
```

When I try to run my script `npm run build:development` which in `package.json` represents:

```json
"scripts": {
	"clean": "rimraf .next",
	"dev": "next dev",
	"export": "next export",
	"start": "next start",
	"lint": "next lint",
	"build:development": "next build && npm run export && npm run rss:development",
	"build": "next build && npm run export && npm run rss",
	"rss:development": "node ./.next/server/scripts/build-rss.js",
	"rss": "node ./.next/serverless/scripts/build-rss.js"
}
```

It throws an error saying:

> Error: Cannot find module '../../webpack-runtime.js'

But I checked. The file does exist.

The blunder is this used to work earlier. Probably few versions ago when my other project used the same combination.

I have made a complete reproduction showcasing the error → https://github.com/deadcoder0904/next-script-rss-error

Just clone it, install it & try the script `npm run build:development` in the terminal to see the error.

**Comment by deadcoder0904** [comment]
@sokra Did anything change in Webpack v5? This used to work. See https://github.com/tailwindlabs/blog.tailwindcss.com/blob/master/next.config.js#L90 for example :)

**Comment by ghost** [comment]
I would offer a temporary solution to you: downgrade Next.js version to `10.2.3`. It works for me.

**Reply by deadcoder0904** [reply]
@quoc1707 I know that, just opened up an issue yesterday https://github.com/vercel/next.js/issues/28111

The reason I can't just downgrade is some of my stuff uses `Script` tag so I got errors. It also gave a few errors regarding other things. Idk what changed though maybe Next or maybe Webpack :(

**Comment by knandraina** [comment]
Hey I have the exact same error. 

my next.config.js looks like that:

```
module.exports = {
  reactStrictMode: true,
  // target: "experimental-serverless-trace",
  
}
```
Did anyone find a way to solve this issue? I can't deploy my blog...

**Reply by deadcoder0904** [reply]
See if [this solves it](https://github.com/vercel/next.js/issues/28111#issuecomment-952794368) :)

**Comment by knandraina** [comment]
Where should I implement it?

Kévin

**Reply by deadcoder0904** [reply]
@knandraina see my question above. my error was regarding `rss` script being not able to run, if that's not the same repro for you then idk :)

**Comment by Weonik-Sim** [comment]
The same error occurred with aws amplify, so I'm sharing it for now.
I was using nextjs 12.1.2 version, but due to cloudformation's template update, the nextjs version set to amplify disappeared.
After setting the version of nextjs again in the build setting of AWS amplify, it was able to work normally.
It may be necessary to specify the version to build in other environments than the AWS amplify environment.
